onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc

```diff
@@ -87,14 +87,14 @@ MIGraphXExecutionProvider::MIGraphXExecutionProvider(const MIGraphXExecutionProv
     : IExecutionProvider{onnxruntime::kMIGraphXExecutionProvider} {
   // Set GPU device to be used
   hipSetDevice(info.device_id);
-  DeviceAllocatorRegistrationInfo default_memory_info(
-      {OrtMemTypeDefault, [](int id) { return onnxruntime::make_unique<HIPAllocator>(id, MIGRAPHX); }, std::numeric_limits<size_t>::max()});
-  allocator_ = CreateAllocator(default_memory_info, device_id_);
+  AllocatorCreationInfo default_memory_info(
+      {[](int id) { return onnxruntime::make_unique<HIPAllocator>(id, MIGRAPHX); }, device_id_});
+  allocator_ = CreateAllocator(default_memory_info);
   InsertAllocator(allocator_);
 
-  DeviceAllocatorRegistrationInfo pinned_memory_info(
-      {OrtMemTypeCPUOutput, [](int) { return onnxruntime::make_unique<HIPPinnedAllocator>(0, MIGRAPHX_PINNED); }, std::numeric_limits<size_t>::max()});
-  InsertAllocator(CreateAllocator(pinned_memory_info, device_id_));
+  AllocatorCreationInfo pinned_memory_info(
+      {[](int) { return onnxruntime::make_unique<HIPPinnedAllocator>(0, MIGRAPHX_PINNED); }, std::numeric_limits<size_t>::max()}, device_id_);
+  InsertAllocator(CreateAllocator(pinned_memory_info));
```

Comment:
This looks off
```
    {
      [](int) { return onnxruntime::make_unique<HIPPinnedAllocator>(0, MIGRAPHX_PINNED); }, 
      std::numeric_limits<size_t>::max()
    }, 
    device_id_);
```

May be simpler to just call the ctor, and the args seem mismatched

Reply:
This style (mixing of aggregate initialization with constructor call) existed even before this PR. Will fix the arg mismatch. Thx for catching it.